src/lib/sidebar/InterventionList.svelte

```diff
@@ -14,13 +14,18 @@
       return feature.properties.planning?.name || "Untitled polygon";
     }
     if (schema == "v2") {
-      return (
-        feature.properties.v2?.Route?.name ||
-        feature.properties.v2?.Crossing?.name ||
-        "Untitled intervention"
-      );
+      if (feature.properties.v2) {
+        if ("Route" in feature.properties.v2) {
+          return feature.properties.v2!.Route.name || "Untitled route";
```

Comment:
I'm still a bit confused. Maybe I'm missing something but I think we're using something that looks exactly like jsonschema (in `v2.json`) and choosing to have it behave differently and I don't know why we're doing that. Why don't we just have `OneOf` work such that `v2` is either a `Route` or a `Crossing` rather than it containing either `Route` or `Crossing` as a member?

Reply:
Any way we express the `OneOf` needs to work with https://www.typescriptlang.org/docs/handbook/2/narrowing.html -- in the code, we need to be able to distinguish whether something is a route, a crossing, or undefined/missing/not filled out yet.

The current approach is `type Intervention = { Route: Route } | { Crossing: Crossing } | {}`. This explicitly nests the `OneOf` case as another dictionary, so the JSON looks like `v2: { Route: { name: "foo", ... } }`. Type narrowing uses the `in` operator.

Alternatively, we could remove the nesting and use a `kind` or `type` field. This'd be something like `type Intervention = Route | Crossing | {}`, where the `Route` interface would include a required field `kind: "Route"`, and the JSON would be something like `v2: { kind: "Route", name: "foo", ... }`. Our [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) in this case would check `properties.v2.kind == "Route"` to do the narrowing. I think this might be the approach you're in favor of?

For reference, https://transform.tools/json-schema-to-typescript is a tool to see one interpretation of expressing the JSON schema in TS. (I don't think there's necessarily one canonical implementation or anything like that)